test/proposal.test.ts

```diff
@@ -85,32 +76,64 @@ describe("proposal", function () {
   });
 
   describe("deploys expected contracts", function () {
-    let gnosisDao: Contract;
-    let contracts: Record<keyof FinalAddresses | "virtualCowToken", Contract>;
+    const cowDaoSettings: SafeCreationSettings = {
+      owners: [1, 2, 3, 4, 5].map((i) => "0x".padEnd(42, i.toString())),
+      threshold: 5,
+    };
+    const teamConrollerSettings: SafeCreationSettings = {
+      owners: [6, 7, 8].map((i) => "0x".padEnd(42, i.toString())),
+      threshold: 2,
+    };
+    const virtualTokenCreationSettings: VirtualTokenCreationSettings = {
+      merkleRoot: "0x" + "42".repeat(32),
+      usdcToken: "0x0000" + "42".repeat(17) + "01",
+      gnoToken: "0x0000" + "42".repeat(17) + "02",
+      gnoPrice: "31337",
+      wrappedNativeToken: "0x0000" + "42".repeat(17) + "03",
+      nativeTokenPrice: "42424242",
+    };
 
+    let settings: DeploymentProposalSettings;
+    let contracts: Record<keyof FinalAddresses | "virtualCowToken", Contract>;
     before(async function () {
+      const bridgeParameters: BridgeParameter = {
+        multiTokenMediatorGnosisChain: "0x" + "01".repeat(20),
+        multiTokenMediatorETH: multiTokenMediatorETH.address,
+        amountToRelay: utils.parseEther("1").toString(),
+      };
+      settings = {
+        gnosisDao: gnosisDao.address,
+        cowDao: cowDaoSettings,
+        teamController: teamConrollerSettings,
+        cowToken: {},
+        virtualCowToken: virtualTokenCreationSettings,
+        bridge: bridgeParameters,
+      };
       const { steps, addresses } = await generateProposal(
         settings,
         gnosisSafeManager.getDeploymentAddresses(),
         hre.ethers,
       );
 
-      gnosisDao = await (
-        await gnosisSafeManager.newSafe([gnosisDaoOwner.address], 1)
-      ).connect(executor);
+      let virtualTokenCreationTxResponse: TransactionResponse;
 
-      let lastResponse: TransactionResponse;
-      for (const step of steps) {
-        lastResponse = await execSafeTransaction(gnosisDao, step, [
-          gnosisDaoOwner,
-        ]);
+      for (const step of deploymentStepsIntoArray(steps)) {
+        if (step == steps.virtualCowTokenCreationTransaction) {
+          virtualTokenCreationTxResponse = await execSafeTransaction(
+            gnosisDao,
+            step,
+            [gnosisDaoOwner],
+          );
+        } else {
+          await execSafeTransaction(gnosisDao, step, [gnosisDaoOwner]);
+        }
```

Comment:
I think this is why the proposal doesn't return an array anymore. Another way to do it without looking at the internals of the proposal generation would be to:
1. Keep a set of all proxies generated in each proposal step.
2. After everything is executed, subtract the set of known proxy deployed (Cow DAO, ...)
3. Assert that only one element remains and use this one as the virtual token.

Reply:
Note that you had to come up with this workaround because I was too lazy to do it myself and used the implicit assumption that the last transaction was the one deploying the virtual token. :sweat_smile: